### PR TITLE
Configuration for body parsing size limit

### PR DESCRIPTION
**Pull request by huv1k**
This PR introduces the possibility to change the size of the parsed body. This brings features like:
- Adjusting the size of the incoming message
- Better security

### CODE REVIEW

File: packages/next/README.md [hunk 1]
```diff
@@ -1114,7 +1114,7 @@ Those middlewares are:
 - `req.query` - an object containing the [query string](https://en.wikipedia.org/wiki/Query_string). Defaults to `{}`
 - `req.body` - an object containing the body parsed by `content-type`, or `null` if no body is sent
 
-Body parsing is enabled by default.
+Body parsing is enabled by default and size limit of parsed body is `1mb`.
```
Comment (lfades) [inline packages/next/README.md]:
```suggestion
Body parsing is enabled by default with a size limit of `1mb` for the parsed body.
```

**Review by lfades — DISMISSED** [review]

File: packages/next-server/server/api-utils.ts [hunk 2]
```diff
@@ -29,14 +30,17 @@ export async function apiResolver(
...
+      }
     }
     // Parsing of cookies
     setLazyProp({ req }, 'cookies', getCookieParser(req))
     // Parsing query string
     setLazyProp({ req, params }, 'query', getQueryParser(req))
     // // Parsing of body
     if (bodyParser) {
-      req.body = await parseBody(req)
+      req.body = await parseBody(req, bodySizeLimit)
```
Comment (timneutkens) [inline packages/next-server/server/api-utils.ts]: Can't this be: 

```suggestion
      req.body = await parseBody(req, config.api && config.api.bodySizeLimit ? config.api.bodySizeLimit : '1mb')
```

File: packages/next/README.md [hunk 3]
```diff
@@ -1,4 +1,5 @@
-[![Next.js](https://assets.zeit.co/image/upload/v1538361091/repositories/next-js/next-js.png)](https://nextjs.org)
+[![Next.js](h
```
Comment (Timer) [inline packages/next/README.md]: Seems this was broken.

File: packages/next-server/types/index.d.ts [hunk 4]
```diff
@@ -5,6 +5,11 @@ export type PageConfig = {
   amp?: boolean | 'hybrid'
   api?: {
     bodyParser?: boolean
+    /**
+     * The byte limit of the body. This is the number of bytes or any string
+     * format supported by `bytes`, for example `1000`, `'500kb'` or `'3mb'`.
+     */
+    bodySizeLimit?: number | string
```
Comment (Timer) [inline packages/next-server/types/index.d.ts]: This should be merged into the `bodyParser` key IMO.

e.g.
```ts
api?: { 
  bodyParser?: { sizeLimit?: number | string } | false 
}

**Review by Timer — CHANGES_REQUESTED** [review]

**Review by Timer — APPROVED** [review]

**Pull request merged**